src/App.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect } from 'react';
+import Dice from './components/Dice';
+import { nanoid } from 'nanoid';
+import Confetti from 'react-confetti';
+
+const App = () => {
+  const generateNewDice = () => {
+    const newDice = [];
+    for (let i = 0; i < 10; i++) {
+      newDice.push({
+        id: nanoid(),
+        value: Math.ceil(Math.random() * 6),
+        isHeld: false,
+      });
+    }
+    return newDice;
+  };
+
+  const rollNewDice = () => {
+    if (!tenzies) {
+      setDice(
+        dice.map(die => {
+          return die.isHeld ? die : { id: nanoid(), value: Math.ceil(Math.random() * 6), isHeld: false };
+        })
+      );
+    } else {
+      setDice(generateNewDice);
+      setTenzies(false);
+    }
+  };
+
+  const handleHold = id => {
+    setDice(
+      dice.map(die => {
+        return die.id === id ? { ...die, isHeld: !die.isHeld } : die;
+      })
+    );
+  };
+
+  const [dice, setDice] = useState(generateNewDice);
+  const [tenzies, setTenzies] = useState(false);
```

Comment:
can you Move this states at the top of the function it will improve the formatting of your code.